Lib/imaplib.py

```diff
@@ -1579,9 +1577,10 @@ def run(cmd, args):
             run(cmd, args)
 
         for ml in run('list', ('/tmp/', 'yy%')):
-            mo = re.match(r'.*"([^"]+)"$', ml)
-            if mo: path = mo.group(1)
-            else: path = ml.split()[-1]
+            if (mo := re.match(r'.*"([^"]+)"$', ml)):
+                path = mo.group(1)
+            else:
+                path = ml.split()[-1]
```

Comment:
The old code could have used the ternary

`path = mo.group(1) if mo else ml.split()[-1]`

However, the replacement for that would now look like

`path = mo.group(1) if (mo := re.match(r'.*"([^"]+)"$', ml)) else ml.split()[-1]`

I think that loses some readability because of the complicated if condition. Using the `if else` statement seems better here.